Fix Task Cancelled exception called from .Net SDK tests

casskir opened the pull request.

casskir committed:
Fix Task Cancelled exception called from .Net SDK tests

farseev commented:
1. Since tests are walking now, we can change 
+dotnet test -p:ParallelizeTestCollections=false 
back to multithread execution to make it the same as in .Net Version (remove -p:ParallelizeTestCollections=false)

2. It would be nice to have more informative preprocessor constant in
#if NET
using bBridgeAPISDKNET.Test;
#endif

"NET" does not immediately mean Windows's .Net
We could also use indent there to emphasize preprocessor directive

3. Similarly to (2), need to rename projects, folders, and solutions to make .Net Core and .Net Windows being easily distinguished; change Readme accordingly

4. Create scripts (locally or in repository) that would generate NUGEts for both .Net Core and .Net Windows projects;

5. Can merge after completed...

casskir committed:
Back to multithread execution to make it the same as in .Net Version
More informative preprocessor constant NET->NETCORE
Projects rename
Create build scripts for .NET Core and .NET Classic

casskir committed:
Fix preprocessor directive

casskir committed:
Revert framework to 1.1

farseev commented:
Good Job! Need to fix some minor issues. No further revision necessary, merge upon completion.

1. Package.nuspec: 
     id: SoMin.API.SDK
     version: 1.0.4
     author: bBridge Team
     owners: bBridge Team
     project Url: http://dev.somin.ai/
     require License Acceptance: true
     description: SoMin API SDK is a .Net library for making calls to SoMin API. The library enables users for making API calls from all .Net-enabled platforms
     release Notes: SoMin.API.SDK for .Net 4.5
     copyright: bBridge Team
     tags: Social Multimedia Analytics
1a. We may need to change namespaces from bBridgeAPISDK to SoMinAPISDK as well in the projects. The SDK is maintained by bbridge open source code team. However, the name of the API has changed from bBridge so SoMin and now maintained by SoMin
2. Same changes for .core nuget specification as above
3. Add a script that creates nuget for .Core to .Core build scipt (Let Travis build it but ignore the built artifact for now)
4. bBridgeAPISDK -> bBridgeAPISDKNETCore.Test shows the rename suggestion since the current namespace does not correspond to file location. Fix namespaces according to locations, Follow Resharper suggestions on this.
5.  Add /// <inheritdoc /> to the classes that inherit some other classes with documentation to make them inherit documentation.
I.e:
    /// <inheritdoc />
    /// <summary>
    /// Image processing API capabilities
    /// </summary>
    public class ImageProcessor : APIRequester

6. Separate preprocessor directives from imports:
using Xunit;
#if NETCORE

using Xunit;

#if NETCORE
7. I like the idea to keep shared libraries in an independent folder. :+1:

casskir committed:
Updated package.nuspec for both project
Added step to make package for NETCore project
Made same name for projects and directories bBridgeAPISDK-NETCore -> bBridgeAPISDKNETCore
Separate preprocessor directives

casskir committed:
Rename project in build scripts
